01-data-structures/03-linked-lists/linked_list.rb

```diff
@@ -4,27 +4,107 @@ class LinkedList
   attr_accessor :head
   attr_accessor :tail
 
+  def initialize
+    @head, @tail, = nil, nil
+  end
+
   # This method creates a new `Node` using `data`, and inserts it at the end of the list.
   def add_to_tail(node)
+    if @head
+      current_node = @head
+      while current_node.next
+        current_node = current_node.next
+      end
+      current_node.next = node
```

Comment:
You could get rid of this loop and just do:

```ruby
if @head
  @tail.next = node
  @tail = node
else
...
```

This should speed up your insertion time and improves the complexity of the insertion from `O(n)` to `O(1)`